src/main/js/App.js

```diff
@@ -118,24 +119,37 @@ const App = () => {
                 'Content-Type': 'application/json'
             },
         })
-        .then(res => res.json())
+        .then(res => {
+            if (res.status === 200) {
+                return res.json();
+            } else {
+                throw res;
+            }
+        })
         .then(function (data) {
             console.log(data);
             if (data.message === "ok") {
                 setConsoleText(consoleText + "\nSurvey " + survey.name + " Created with ID: " + data.id);
             } else {
                 setConsoleText(consoleText + "\nSurvey Creation Error: " + data.content);
             }
+            // Retrieve survey names after creating a new survey
+            retrieveSurveyNames();
         })
         .catch(console.log);
     };
 
     const retrieveSurvey = () => {
-        console.log(`${webUrl}retrieveSurvey?id=${userSurveyName}`);
-        fetch(`${webUrl}retrieveSurvey?id=${userSurveyName}`)
-        .then(res => res.json())
+        console.log(`${webUrl}retrieveSurvey?id=${userSurveyId}`);
```

Comment:
We shouldn't leave console.logs in the code

Reply:
I concur on this stipulation